interfaces/builtin/docker_support.go

```diff
@@ -161,12 +161,172 @@ pivot_root,
 # use 'privileged-containers: true' to support --security-opts
 
 # defaults for docker-default
-change_profile unsafe /** -> docker-default,
+# Unfortunately, the docker snap is currently (by design?) setup to have both 
+# the privileged and unprivileged variant of the docker-support interface 
+# connected which means we have rules that are compatible to allow both 
+# transitioning to docker-default profile here AAAAAAND transitioning to any 
+# other profile below in the privileged snippet, BUUUUUUUT also need to be 
+# triply compatible with the injected compatibility snap-confine transition 
+# rules to temporarily support executing other snaps from devmode snaps. 
+# So we are left with writing out these extremely verbose regexps because AARE 
+# does not have a negative concept to exclude just the paths we want. 
+# See also https://bugs.launchpad.net/apparmor/+bug/1964853 and
+# https://bugs.launchpad.net/apparmor/+bug/1964854 for more details on the 
+# AppArmor parser side of things.
+# TODO: When we drop support for executing other snaps from devmode snaps (or 
+# when the AppArmor parser bugs are fixed) this can go back to the much simpler
+# rule:
+# change_profile unsafe /** -> docker-default,
+# but until then we are stuck with:
+change_profile unsafe /[^s]** -> docker-default,
+change_profile unsafe /s[^n]** -> docker-default,
+change_profile unsafe /sn[^a]** -> docker-default,
+change_profile unsafe /sna[^p]** -> docker-default,
+change_profile unsafe /snap[^/]** -> docker-default,
+change_profile unsafe /snap/{s[^n],c[^o]}** -> docker-default,
```

Comment:
Do we need a line above this of the form:

```
 change_profile unsafe /snap/{[^s],[^c]}** -> docker-default,
```

Reply:
Yes you're right we should, I missed this when originally writing these rules

Reply:
done